### PR TITLE
Add spring code sample

### PR DESCRIPTION
**Pull request by gurbuzali**
- Add annotation based configuration of Jet
- Add xml based configuration of Jet (using built-in spring bean/property xml elements for Jet specific configurations and `hz` schema for configuring IMDG specific configurations)
- Add `@SpringAware` annotated custom source processor sample

### CODE REVIEW

File: spring/pom.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    <version>0.6-SNAPSHOT</version>
+
+
+    <properties>
+        <!-- needed for checkstyle/findbugs -->
+        <main.basedir>${project.parent.basedir}</main.basedir>
+
+        <spring.version>4.2.3.RELEASE</spring.version>
+        <hazelcast.spring.version>3.10-SNAPSHOT</hazelcast.spring.version>
```
Comment (cangencer) [inline spring/pom.xml]: will this only work with a specific version of Hazelcast? We should add a note here.

File: spring/src/main/java/jet/spring/CustomSourceP.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+        @Nonnull
+        @Override
+        public Function<Address, ProcessorSupplier> get(@Nonnull List<Address> addresses) {
+            return a -> c -> range(0, c).mapToObj(i -> new CustomSourceP()).collect(toList());
+        }
+
+        @Override
+        public int preferredLocalParallelism() {
+            return 1;
```
Comment (cangencer) [inline spring/src/main/java/jet/spring/CustomSourceP.java]: you can use ProcessorMetaSupplier.dontParallize here and just have `ProcessorMetaSupplier.of(CustomSourceP::new)`

File: spring/src/main/java/jet/spring/CustomSourceP.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+public class CustomSourceP extends AbstractProcessor {
+
+    private Traverser<User> traverser;
+
+    @Autowired
+    private transient UserDao userDao;
+
+    @Override
+    protected void init(@Nonnull Context context) {
+        traverser = Traversers.traverseIterable(userDao.findAll());
```
Comment (cangencer) [inline spring/src/main/java/jet/spring/CustomSourceP.java]: is this blocking? should be marked as not cooperative in this case.

**Comment by cangencer** [comment]
I think some of the classes could use a bit more explanation through comments.

File: spring/pom.xml (hunk 1 above)
Comment (gurbuzali) [reply spring/pom.xml]: I've tried with 3.9 and below versions, 3.9 runs fine but below that we are getting class/method not found exceptions. So yes, user should use specific version of hazelcast, preferably the same version which jet shades.

**Pull request merged**